lib/platform/gitea/gitea-helper.spec.ts

```diff
@@ -766,22 +767,104 @@ describe('platform/gitea/gitea-helper', () => {
     });
 
     it('should properly determine worst commit status', async () => {
-      const statuses: ght.CommitStatusType[] = [
-        'unknown',
-        'success',
-        'pending',
-        'warning',
-        'failure',
-        'error',
+      const statuses: {
+        status: ght.CommitStatusType;
+        created_at: string;
+        expected: ght.CommitStatusType;
+      }[] = [
+        {
+          status: 'unknown',
+          created_at: '2020-03-25T01:00:00Z',
+          expected: 'unknown',
+        },
+        {
+          status: 'success',
+          created_at: '2020-03-25T02:00:00Z',
+          expected: 'success',
+        },
+        {
+          status: 'pending',
+          created_at: '2020-03-25T03:00:00Z',
+          expected: 'pending',
+        },
+        {
+          status: 'warning',
+          created_at: '2020-03-25T04:00:00Z',
+          expected: 'warning',
+        },
+        {
+          status: 'failure',
+          created_at: '2020-03-25T05:00:00Z',
+          expected: 'failure',
+        },
+        {
+          status: 'error',
+          created_at: '2020-03-25T06:00:00Z',
+          expected: 'error',
+        },
+        {
+          status: 'failure',
+          created_at: '2020-03-25T07:00:00Z',
+          expected: 'failure',
+        },
+        {
+          status: 'warning',
+          created_at: '2020-03-25T08:00:00Z',
+          expected: 'warning',
+        },
+        {
+          status: 'pending',
+          created_at: '2020-03-25T09:00:00Z',
+          expected: 'pending',
+        },
+        {
+          status: 'success',
+          created_at: '2020-03-25T10:00:00Z',
+          expected: 'success',
+        },
+        {
+          status: 'unknown',
+          created_at: '2020-03-25T11:00:00Z',
+          expected: 'unknown',
+        },
+        {
+          status: 'success',
+          created_at: '2020-03-25T02:00:00Z',
+          expected: 'unknown',
+        },
+        {
+          status: 'pending',
+          created_at: '2020-03-25T03:00:00Z',
+          expected: 'unknown',
+        },
+        {
+          status: 'warning',
+          created_at: '2020-03-25T04:00:00Z',
+          expected: 'unknown',
+        },
+        {
+          status: 'failure',
+          created_at: '2020-03-25T05:00:00Z',
+          expected: 'unknown',
+        },
+        {
+          status: 'error',
+          created_at: '2020-03-25T06:00:00Z',
+          expected: 'unknown',
+        },
```

Comment:
Is the order guaranteed? Or should we shuffle?

Reply:
We can shuffle manually to add more case,
but not programatically becuase expected value is based on previous added statuses,

Reply:
i think simply reorder would be enough for this test.  so no more tests required